src/main/java/io/kestra/plugin/couchbase/Query.java

```diff
@@ -0,0 +1,136 @@
+import lombok.experimental.SuperBuilder;
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import java.io.*;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Query a Couchbase database with N1QL."
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Send a N1QL query to a Couchbase Database",
+            code = {
+                "connectionString: couchbase://localhost",
+                "username: couchbase_user",
+                "password: couchbase_passwd",
+                "query: SELECT * FROM COUCHBASE_BUCKET(.COUCHBASE_SCOPE.COUCHBASE_COLLECTION)",
+                "fetchType: FETCH"
+            }
+        ),
+    }
+)
+public class Query extends CouchbaseConnection implements RunnableTask<Query.Output>, QueryInterface {
+    @NotNull
+    @Builder.Default
+    protected FetchType fetchType = FetchType.NONE;
+    protected Object parameters;
+
+    @NotNull
+    @NotBlank
+    protected String query;
+
+    public Output run(RunContext runContext) throws Exception {
+        Cluster session = connect(runContext);
+
+        String renderedQuery = runContext.render(query);
+        QueryOptions parametersForQuery = getParametersForQuery();
+        QueryResult result = session.query(renderedQuery, parametersForQuery);
+
+        close(session);
+
+        List<Map<String, Object>> rowsAsMap = result.rowsAs(new TypeRef<>() {
+        });
+
+        Output.OutputBuilder outputBuilder = Output.builder().size((long) rowsAsMap.size());
+        switch (fetchType) {
+            case FETCH: {
+                return outputBuilder
+                    .rows(rowsAsMap)
+                    .build();
+            }
+            case FETCHONE: {
+                return outputBuilder
+                    .row(rowsAsMap.stream().findFirst().orElse(null))
+                    .build();
+            }
+            case STORE: {
+                File tempFile = runContext.tempFile(".ion").toFile();
+                BufferedWriter fileWriter = new BufferedWriter(new FileWriter(tempFile));
+                try (OutputStream outputStream = new FileOutputStream(tempFile)) {
+                    FileSerde.write(outputStream, rowsAsMap);
+                }
+
+                fileWriter.flush();
+                fileWriter.close();
+
+                return outputBuilder
+                    .uri(runContext.putTempFile(tempFile))
+                    .build();
+            }
+        }
+
+        return outputBuilder.build();
+    }
+
+    private QueryOptions getParametersForQuery() {
+        QueryOptions queryOptions = QueryOptions.queryOptions();
+
+        if (parameters instanceof Map) queryOptions.parameters(JsonObject.from((Map<String, ?>) parameters));
+        else if (parameters instanceof List) queryOptions.parameters(JsonArray.from((List<?>) parameters));
```

Comment:
Please always use {} even for one-line if/else this avoid future bugs when someone add a line to one of the branch